components/ingest-service/pipeline/publisher/msg_distributor_test.go

```diff
@@ -0,0 +1,122 @@
+package publisher
```

Comment:
nit: By convention, this file should be called `msg_distributor_internal_test.go` since it is testing the internals of the package.

Reply:
Thouuuuugh, thanks for adding tests!!!!! 💯